klient/constructor/meta/meta_constructor.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package meta contains constructor type to build values of type *coreV1.ObjectMeta
+package meta
+
+import (
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	EmptyName = ""
+	EmptyLabels = map[string]string{}
+	DefaultNamespace = "default"
+    ObjectMetaNone = ObjectMetaConstructor{obj: metaV1.ObjectMeta{}}
+)
+
+// ObjectMetaConstructor constructor for type coreV1.ObjectMeta
+type ObjectMetaConstructor struct {
+	obj metaV1.ObjectMeta
+}
+
+// Object is the initializer function for ObjectMetaConstructor
+func Object(name string) ObjectMetaConstructor {
```

Comment:
Can we provide some kind of support for `GenerateName` 

If we turn the `Object` into a no-args constructor, then we can provide a `.Name` and `.GenerateName` chained API calls to let user decide what they want to use ? It comes really handy to be able to generate pod names and resource names in cases where you are generating a bunch of pods/resources to load up the system for some test and you don't really care much about the name of each of them.

Reply:
@harshanarayana Yes. Ultimately, the API should provide similar behavior as what the API server does with defaults. Such as generate names (where appropriate) and provide sensible defaults where needed.